### ISSUE TITLE
[issue] should the mojo toolchains be marked as `dev_dependency`?

### ISSUE THREAD

**Issue by albertocavalcante** [modular/rules_mojo#22]
Adding `rules_mojo` as a dependency causes Bazel to fail, even when not using any Mojo rules:

```starlark
bazel_dep(name = "rules_mojo", version = "0.4.1", dev_dependency = True)
```

This triggers the following error:

```sh
ERROR: Analysis of target '//src/parser:parser' failed; build aborted: module extension @@rules_mojo+//mojo:extensions.bzl%mojo does not generate 
repository "mojo_gpu_toolchains", yet it is imported as "mojo_gpu_toolchains" in the usage at 
https://bcr.bazel.build/modules/rules_mojo/0.4.1/MODULE.bazel:12:21 (did you mean 'mojo_toolchains'?)
```

Today `rules_mojo` `MODULE.bazel` has:

```starlark
mojo = use_extension("//mojo:extensions.bzl", "mojo")
mojo.toolchain()
mojo.gpu_toolchains()
use_repo(mojo, "mojo_gpu_toolchains", "mojo_host_platform", "mojo_toolchains")

register_toolchains("@mojo_toolchains//...", "@mojo_gpu_toolchains//...")
```

If I also register this toolchain in my project, it works. I also noticed that Modular (https://github.com/modular/modular/blob/main/bazel/common.MODULE.bazel) registers in a slightly different way, but ultimately, having the toolchain resolves the issue.

Still, simply adding the rule as a dependency shouldn't trigger the expectation that a toolchain is also available. I wonder if adding `dev_dependency` is either necessary or would solve this. I've seen `rules_go` does not set `go_sdk` as `dev_dependency`, for example, so it makes me wonder.
